backend/src/layer/data/flickr_api/json_parser.rs

```diff
@@ -0,0 +1,234 @@
+use crate::interface::image_hoster::model::ImageMetaData;
+use crate::interface::image_hoster::ImageHosterError;
+use crate::layer::data::flickr_api::json_structs::*;
+
+pub struct JSONParser;
+
+fn get_selected_size_fallback() -> String { String::from("Medium") }
+fn get_selected_size() -> String { String::from("Large") }
+
+impl JSONParser {
+    pub const fn new() -> Self {
+        Self
+    }
+
+    /// Obtains the preferred image information from the [`JsonRootSizes`] struct.
+    /// # Return
+    /// The [`ImageMetaData`] struct containing all necessary information for the image.
+    /// If the preferred size cannot be obtained a fallback to a smaller size 'll be done.
+    /// If even this fallback size is not available the url will be empty. No url 'll be provided.
+    pub fn parse_get_sizes(&self, root: JsonRootSizes, photo_id: &str) -> ImageMetaData {
+        let mut url = String::new();
+        for size in root.sizes.size.clone() {
+            if size.label == get_selected_size() {
+                url = size.source;
+                break;
+            }
+        }
+        if url == String::new() {
+            for size in root.sizes.size {
+                if size.label == get_selected_size_fallback() {
+                    url = size.source;
+                    break;
+                }
+            }
+        }
+        ImageMetaData {
+            id: String::from(photo_id),
+            image_url: url,
+        }
+    }
+
+    /// Obtains and validates the license by the information from the [`JsonRootLicense`] struct.
+    /// # Return
+    /// A boolean if the image has an valid license or not.
+    /// If the image has no license or no license history, the image isn't restricted by any license and true 'll be returned.
+    pub fn check_license(&self, root: JsonRootLicense) -> bool {
+        let mut last_date: u64 = 0;
+        for entry in root.rsp.license_history.clone() {
+            if last_date < entry.date_change {
+                last_date = entry.date_change;
+            }
+        }
+        let mut license: String = String::new();
+        for entry in root.rsp.license_history {
+            if entry.date_change == last_date {
+                license = entry.new_license;
+            }
+        }
```

Comment:
```suggestion
        let license: String = root.rsp.license_history.iter().max_by_key(|l| l.date_change).unwrap_or_default();
```
it can be so easy...

Reply:
so the licenses are not always sorted?